thunder/core/update_aliases.py

```diff
@@ -42,11 +51,15 @@ def _involves_viewed_args(bsym, viewed):
     return any(isinstance(p, TensorProxy) and variableify(p) in viewed for p in bsym.flat_proxy_args)
 
 
-def insert_alias_updates(computation_trace: Trace) -> Trace:
+def insert_alias_updates(computation_trace: Trace, alias_tensor_indices: list[list[int]]) -> Trace:
     swap_map = dict()
     bsyms = []
 
-    # First pass: identify views, their originals, and operands involved in inplace ops
+    # First pass: identify inputs which are views of each other and swap them out with a default,
+    # reshaping if necessary.
+    computation_trace, _ = replace_args_with_alias_map(computation_trace, alias_tensor_indices)
```

Comment:
Now the first step of update_aliases is the same as with in-place functionalization.